pkg/function/backup_data.go

```diff
@@ -134,10 +134,10 @@ func backupData(ctx context.Context, cli kubernetes.Interface, namespace, pod, c
 		return "", "", err
 	}
 	defer cleanUpCredsFile(ctx, pw, namespace, pod, container)
-	if err = restic.GetOrCreateRepository(cli, namespace, pod, container, backupArtifactPrefix, encryptionKey, tp.Profile); err != nil {
+	encryptionKey, err = restic.GetOrCreateRepository(cli, namespace, pod, container, backupArtifactPrefix, encryptionKey, tp.Profile)
```

Comment:
@DeepikaDixit - why are we returning `encryptionKey`?

Reply:
We need to `revert` to `default encryptionKey` for all calls in the `func`
If we don't return the next call at line 143 `restic.BackupCommandByTag(tp.Profile, backupArtifactPrefix, backupTag, includePath, encryptionKey)` will still use the encryptionKey which is not correct against current repo.